src/Symfony/Component/Console/Application.php

```diff
@@ -985,7 +985,7 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
 
         if ($this->signalsToDispatchEvent) {
             $commandSignals = $command instanceof SignalableCommandInterface ? $command->getSubscribedSignals() : [];
-            $dispatchSignals = $this->dispatcher && $this->dispatcher->hasListeners(ConsoleEvents::SIGNAL);
+            $dispatchSignals = null !== $this->dispatcher;
```

Comment:
let's drop the variable and `if ($this->dispatcher) `?

Reply:
done